peer/peer.go

```diff
@@ -775,6 +784,7 @@ func (p *Peer) PushAddrMsg(addresses []*wire.NetAddress, subnetworkID *subnetwor
 //
 // This function is safe for concurrent access.
 func (p *Peer) PushGetBlockLocatorMsg(highHash, lowHash *daghash.Hash) {
+	p.shouldSendBlockLocator = true
```

Comment:
I'd rename to `isBlockLocatorPending` or `wasBlockLocatorRequested` something.
Because shouldSendBlockLocator sound like "should we send blockLocator".

Also, I don't see anywhere that you set it to false. Shouldn't this happen on handling block locator or something?